### DISCUSSION TITLE
New RPM sensor using AMR magnetic sensor

### DISCUSSION THREAD

**Discussion by DG-PROPULSION** [Jehanzeb1973/Ardu_ECU#8] [Ideas]
Hi @Jehanzeb1973 
I have made my own RPM sensor using commercially available AMR magnetic sensors.
I designed my schematic and ordered the PCBs. The good thing is the sensors are working.

BUT the issue is this sensor responds to both NORTH and SOUTH of the magnet present in the compressor nut. Hence it shows 2x  RPM than the actual RPM.
Is there are way to edit the code to count half the pulses every rotation or divide the RPM sensor output by 2?

Thanks.

**Comment by Jehanzeb1973** [comment]
Hi dstromeng
That’s great news. This is absolutely possible. RPMAvg global variable in ReadRPM() function stores the results of RPM measurement. You can divide this by 2 to half the measurement.

Regards

Jehanzeb Nurzai
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/




From: dstromeng
Sent: Friday, 9 June 2023 6:21 am
To: Jehanzeb1973/Ardu_ECU
Cc: Jehanzeb Khan; Mention
Subject: [Jehanzeb1973/Ardu_ECU] New RPM sensor using AMR magnetic sensor(Discussion #8)

Hi @Jehanzeb1973
I have made my own RPM sensor using commercially available AMR magnetic sensors.
I designed my schematic and ordered the PCBs. The good thing is the sensors are working.
BUT the issue is this sensor responds to both NORTH and SOUTH of the magnet present in the compressor nut. Hence it shows 2x RPM than the actual RPM.
Is there are way to edit the code to count half the pulses every rotation or divide the RPM sensor output by 2?
Thanks.

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
Yea I figured that out after going through the code.
Almost every logic in the code works on RPMavg.

I tested the ECU on one of my engines today. It resulted in sort of a runaway condition and my turbine end bearing got jammed. The engine was a 18kg thrust engine Jet Central Rhino gas start version.

What settings do you usually have to change to adapt the ECU to other engines?

I changed-
Max RPM to 115000
Gas On RPM to 1000
Starter OFF RPM to 27000
Starter limit to 0 (since I am using s brushed starter with brushed ESC)

Max starter voltage to 7.8
Max Fuel pump voltage to 7.8 Since both of them are running on 2s lipo.

Thanks

**Comment by Jehanzeb1973** [comment]
Hi dstromeng

The key setting to prevent runaway is 
Acceleration Delay (millisec)   60 
Low RPM Acceleration Delay (millisec) 600


This is the delay in milliseconds between each iteration of fuel pump increase. This is because there is a delay between fuel pump increment and RPM increment. If fuel pump is increased quickly, the engine gets flooded before it has time to accelerate and consume the fuel. 
I would suggest two things for trial.
1- Reduce Max RPM to 70000 ( this will cut the fuel supply before it accelerates to dangerous RPM.
2- Increase acceleration delay to 600  or even higher- This is the number of milliseconds between each step increase in fuel pump signal. If the acceleration is too slow, then you can reduce this value or if acceleration is still high, you can further increase this value.

 Once you get a stable increase to 70000 RPM, then you can change the Max RPM value and optimize acceleration delay

Jehanzeb Khan
DIGIPAK Pvt. Limited


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
HI @Jehanzeb1973 
Should I also change the Low RPM Acc Delay as well since that's what controls the fuel flow at startup. Do you have any recommendation for that number?

Also I had a few questions about some parameters-
1. **Pump On Value**- What exactly is this parameter? Should it be set to when the pump actually starts spinning. I have seen in my case fuel pump turns ON at about 87. Should I set to that?

2. **Ignition Threshold**- Now I understand this parameter from what it primarily does that in order to detect ignition the temp should be above this temperature. BUT let's assume if for some reason the engine had a flameout but the exhaust temperature is still above this temperature, will the ECU WAIT for the exhaust temperature to get BELOW this ignition threshold temperature in order to start new cycle? One thing I noticed when this happened that the ECU won't start the cooldown cycle to bring the engine exhaust temperature down. I had a flameout during the startup and the exhaust temperature was still at 400c. I had to wait for the temperature to drop below 100c in order for attempting a new startup.
 
3. **Trial Mode Fuel Flow**- Can this parameter somehow cause flooding of the engine during the startup?

4. **Starter Limit**- Should I set that to 0 since I am using a brushed motor. BUT my question is how does the ECU know when it's a brushed or a brushless motor since both of them are controlled through an ESC which just takes a 5v PWM Servo signal. In that case why do we use 0 for brushed motor and a number for brushless motor?

Regards.

**Comment by Jehanzeb1973** [comment]
Hi dstromeng
See my answers below
Regards

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 

Please also shed some light on what does this parameter do and where to set it at?

**Starter Limit**- Should I set that to 0 since I am using a brushed motor. BUT my question is how the ECU knows when it's a brushed or a brushless motor since both of them are controlled through an ESC which just takes a 5v PWM Servo signal. In that case why do we use 0 for brushed motor and a number for brushless motor?

**Reply by DG-PROPULSION** [reply]
@Jehanzeb1973 

Also how do you think setting a too low "Pump On value" can flood the engine?
I checked mine was set at default 25 whereas my pump starts spinning at 87 when this engine flooding happened.

Thanks again

**Reply by DG-PROPULSION** [reply]
@Jehanzeb1973 

Regarding the **Starter Limit**- Should I set that to 0 since I am using a brushed motor. BUT my question is how the ECU knows when it's a brushed or a brushless motor since both of them are controlled through an ESC which just takes a 5v PWM Servo signal. In that case why do we use 0 for brushed motor and a number for brushless motor?

**Comment by Jehanzeb1973** [comment]
I don’t know the exact reason but I observed that when I keep pump on value just below the start off, the engine started smooth. If I keep it too far below, the system kind of builds a momentum and I get sudden acceleration. This was probably due to acceleration delay being too short in my case. With acceleration delay as a large number it should be ok to keep it well below pump on value. You just have to wait longer for engine to get the fuel and start accelerating.
Regards

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Comment by Jehanzeb1973** [comment]
Hi dstromeng
Starter Limit is the max control signal you will give to Starter. ECU cannot detect the type of motor and depends on value of Starter Limit to decide starting logic. The starting logic is different for Brushed motor and Brushless motor.
I have used both Brushed and brushless motors for starter. With brushed motors, there is no magnetic noise and RPM signal is correct. I have seen high magnetic noise with brushless motor and this corrupts the RPM signal. In general RPM reads high values. The question now is how do you know you have reached Idle RPM if your RPM signal is wrong. The way I solved this problem is that I didn’t depend on RPM with brushless motor. I give it a pre set throttle (Starter Limit)  and then stop the starter completely. The trick is to make sure this throttle is high enough that your engine is at idle RPM at that speed of starter motor. Once the motor stops , the magnetic interference is gone and your RPM signal is correct.  If you set Starter limit too low with brushless motor, engine will not increase to idle RPM and temperature will rise to Max temperature. 
REgards

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Thanks @Jehanzeb1973 for clarifying this. I will use starter limit = 0 for my brushed motor then.

**Comment by DG-PROPULSION** [comment]
Hi @Jehanzeb1973 
1. I am slightly confused with **how you prime the pump** before the test run? is it done with the button by double-clicking it? Can you explain how to do that?

2. Also is there a way we can just run the starter motor in case of high temperatures when there is no combustion going on just to Cooldown the unit rapidly? This will be beneficial for the longevity of the bearings.

3. Is there a "**Cooldown**" function built in?
I see there's a cooldown thing built into the waiting.  But you said we cannot cooldown the turbine ina flameout condition and the turbine will have to cool itself down below ignition threshold to start new cycle.

what does this do-

_void WaitingFunction()
{
  errorCode=0; //reset error code
  startStageTimeOld=0;
  gasOnTime=0;
  glowPower=outMin;
  gasFlow=false;
  fuelFlowTarget=outMin;//This will change in below command if system is in pumpPrime mode
 **if(!pumpPrime) startMotorTarget=(rcThrottleSignal*outMax/100);//helps to manually cooldown turbine after abort**
else fuelFlowTarget=(rcThrottleSignal*outMax/100);//helps to prime pump in beginning
if ((rcModeSignal>80)&&(rcThrottleSignal<10)&&(exTemp<ignitionThreshold))//Start only when throttle is below 10%
{
  pumpPrime=false;//reset throttle control at startup to starter control
  engineMode=modeStarting;
  startStage=startStage0;
}_


Thanks.

**Comment by Jehanzeb1973** [comment]
1- Yes, you can double click to cycle through different modes. One of them is pump prime mode
2- System does a cool down if the engine is shut down normally. If it aborts due to an error , it doesn’t do a cool down. In this case, you can reset the power by cycling the power to the board or reset the error by long press of button and then starter will be controlled by throttle.

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Comment by DG-PROPULSION** [comment]
Hi @Jehanzeb1973 

I did some bench HIL testing of the ECU without the engine. Now this time adjusted the Fuel On value to something really close when the fuel pump actually starts pumping. I also increased the Low Acc Delay and the Acceleration delay both.

I still think whenever the fuel pump starts during the spool-up it's pumping just too much fuel into the engine.
How do you think we can decrease this fuel amount going into the engine during the idling?

**Do you think we can decrease the pump increment here in the ControlOutput() function and that will help?**
for example if we change to fuelflowtarget=fuelflowNow+0.25 for 0.25 increments.


_void ControlOutput()
{
 //Check Limits and stop runaway increment or decrement condition
 if (fuelFlowTarget>outMin)
 {
  **if (fuelFlowTarget>(fuelFlowNow+1)) fuelFlowTarget=fuelFlowNow+1;// never exceed more than One point**
  if (fuelFlowTarget<(fuelFlowNow-1)) fuelFlowTarget=fuelFlowNow-1;
 }
 if (startMotorTarget>outMin)
{
  if (startMotorTarget>(startMotorNow+1)) startMotorTarget=startMotorNow+1;
  if (startMotorTarget<(startMotorNow-1)) startMotorTarget=startMotorNow-1;
}_


Regards.

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
The servo library that I am using does not take fractions. 
I have defined outMin and outMax as lower and higher limit of values. By default it is 0-500 so you can increment servo value in 1/500 intervals. You can define it  say 0-1000 to make it a much fine control but I don’t think that is the problem.
My thinking is that during the spool up to idle, a balance needs to be achieved between starter speed and fuel flow. If fuel is excessive and there is not enough air, you will get a wet start or a fuel cutoff due to excess temperature. The solution to this is higher starter speed, and faster starter response. During the run up to idle, starter is also functioning to cool the engine by spinning the compressor at high enough RPM. If you have ignition and flames are coming at the back say due to slightly higher fuel flow, the best solution is to speed up the starter so the excess air consumes the fuel as well as cool down the exhaust. This is easier said than done. I have burned a dozen starter motors trying to find the motor that is powerful enough and light enough to be used. I am using 2700 kv 2212 brushless motor for J66 engine. At 11.1 v which is 3S LiPo, it can give about 30000 RPM. My engine becomes self sustaining at 20-22000 RPM so I have a decent margin from this motor to start the engine.. Even then, I have failed to start if the battery is not fully charged.
If you have the right sized motor , you can do two things to speed up the starter
1- Reduce the starterIncDelay from page1 . This will make starter respond quicker to temperature changes
2- If the ESC And motor allows, use a bigger battery. I.e. say go from 3S LiPO to 4S LiPO
Remember , if you see that fuel flow is automatically reduced due to high exhaust temperature then your starter is not responding fast enough or does not have enough power.
If none of this works, you can try using a needle valve from a nitro RC engine between the pump output and Jet engine input. With this you can reduce the flow. But this will load the pump for the full range of flow.
REgards

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
So I did a lot of tests with the ECU and the engine. BUT none of them was successful.

In almost every test the engine was getting flooded with lots of flames coming out of the exhaust end.

i was using
*Acceleration delay* = 100 milliseconds 
*Low speed acceleration delay* = 1200,1500,2200 milliseconds (did 3 different tests)

Almost every single time engine resulted in fuel flooding.

Now which parameters should I be changing to avoid that. I did change low speed acceleration delay but it did not make any progress.

NOW after which RPM acceleration delay takes over Low speed acceleration delay. Should I also change the acceleration delay to 600 from 60(default)? 

I thought that was only required for when the engine was running?

Also you mentioned speeding up the starter RPM? How can you speed up the starter motor? What is the parameter for that?

Thanks.

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
 Lets go over the startup requirements one by one.

1-Is the starter motor able to deliver the required RPM ?
The way to check this is set the engine up as you would for a start but do not open the fuel valve. Give the maximum throttle and note the RPM.
Starter must be able to give at least self sustain RPM to engine. For KJ66 that I have , self sustain RPM is around 22000 so starter needs to deliver say 26000 RPM for engine to start. It will not start with a motor that delivers 12000 RPM and it may start with a motor that reaches 20-22000 RPM. Having the right starter motor is a must otherwise ECU will pump fuel but starter will not deliver the RPM and you will get flames and exceed maxTemp at exhaust. For larger engines usually self sustain RPM is lesser than 22000 and for smaller engines, it is generally higher than 22000

2- Fuel pump is of correct capacity and not too big. Generally for KJ66 fuel pump with 1000ml / min at full speed is the right size. Check your pump capacity by using pumpPrime mode and using the same battery as you use for start up and run it at full speed and see how much quickly it fills a 1 litre bottle.  You can use say a pump with 2000ml/min capacity  but remember every change in control value will result in double the fuel increase. If it is of a large capacity you can tackle this by giving more time for starter to react and RPM to stabilize. You have used 2200 but don’t be afraid to use larger values for testing. I am currently using 2600 on my J66. You can go to 5000 or even 10000. This only means that ECU will wait 5sec or 10 sec before increasing the fuel step. The only draw back is that starter will run longer and put more load on battery. If you have battery with good capacity, this should not be an issue
3- Keep starter delay low so it can react fast to temperature changes. Keep starterDelay as 10 so it can react quickly

Acceleration delay does not affect startup as it is applied after engine RPM is higher than Idle RPM so you can leave it at say 100.

In summary three actions.
1-Ensure Starter motor can give at least 26000 RPM to dry engine
2- Keep Low speed acceleration delay at 4000 for this test but you can increase it if still too much fuel is pumped
3- Make Starter delay as 10

Do the test with these changes and let me know. Also share a vide if you can of your test for me to see what is happening.

Regards


Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 

i am using a jet central rhino 170 engine for all these tests. It has the original starter motor and pump that it came with.
I have checked the starter motor with full battery voltage of 2s lipo and it ramps up the engine upto 10000 rpm dry.

Also it has its own original ECU.
So in order to set a baseline I did many test runs using the original ECU the same day so everything was working and sized properly as it should.

Now my original starter motor got blown after repeated test runs. It’s a brushed dc motor and it’s windings are burnt.

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
Each ECU that comes with a commercial engine is setup properly for that engine. 
A generic ECU like Ardu_ECU has all the settings to optimize it for different type of engines however these need to be determined through trial. 
I burned the original motor for my engine like you because I ran it too long and it overheated. With original ECU it powered up at right time only for a few seconds so it never over heated. 
For this reason I started using brushless motor as it had much higher power capacity. Now I have optimized the settings for brushed motor as well however these are specific to my J66 engine. You can see my settings for j66 in the attached file. I had a few good startups with these settings. Compare these with your settings to see if there is major difference.
I am almost finished testing the Firmware 11 which will have SD recording as well as more safety built-in to avoid excessive fuel. 

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
Do you have to calibrate the full servo range on the ESCs that we use for the starter motor and the fuel pump?

the reason I am asking this is because some ESCs have to programmed for the full range of the throttle and some are by default Forward-reverse and brake mode.

I noticed this with one of the starter motor ESC I was using that during the failed startup my starter motor would start rotating in reverse direction.

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
I have not programmed for calibration option in current versions of ECU. 
There are three actions done on servo signals.
For input :
I have done is limited the low value of input to 1000us so if your radio is sending servo signal from say 800us to 2400 us. ECU will ignore 800-999us part and only act when it sees 1000us minimum pulse width. So at very low range you might not see any action. No clipping is done at high range

Out put range is defined by two variable pairs
outMin-outMax  is for internal ECU calculation and 
and MIN_MICROS – MAX_MICROS is out put servo signal width

e.g. outMin-outMax of 0-500 means each increment or decrement is 1/500 of full scale in ECU logic
This is then mapped to MIN_MICROS – MAX_MICROS which is set as 1000-2400 us by default

If you change outMax to 100 then the response will be quicker but each increment or decrement will be coarse value. Similarly if you set outMax as 1000 then ECU will respond slower as now it has to go through smaller steps but you can have fine control. Other than this there is no other calibration done on input or output servo signals.
The reversing option is most likely ESC specific and it may be possible to disable it by setting jumpers on ESC. It is however strongly recommended to use braking as that helps to disengage Bendix clutch.
Also FYI. I have firmware 11 almost ready and I think is better in terms of limiting excess fuel and also it allows to record all data on sd card for analysis. It has different pinouts though due to SD card usage. I have used both brushed and brushless motor with it to get smooth startups the very first time. I just need to do a little bit more testing and then it should be out

Regards
 

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
I would definitely want to test the Firmware 11. looking forward to it's release.

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
How's the progress coming up with Firmware 11?
What has changed when it comes to the startup.

Thanks.

**Comment by DG-PROPULSION** [comment]
Hi @Jehanzeb1973 
Thanks for releasing the Firmware 11.
How is your readymade PCB board different from the one that uses a ESP32 Dev-kit module?

I see your board has 3 jumpers that you have to use for different functions BUT the schematic doesn't have any of that. How is that taken care of in the schematic that you posted?

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
Done. Its released now. Its a very solid firmware and startups are fast and smooth. Do the tests and let me know if you find anything that can be changed or improved. I have uploaded on GitHub and You Tube is being updated now.
Regards
Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
This is described in detail in the ECU Rev 11 guide. This situation arises because we want to use SD card in 4 wire mode for higher speed writing. This causes a problem with ESP32 accessing the flash hence programming while the circuit is setup for sd card is a problem. Read the guide and hopefully it will make it clear.
Regards

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
Thanks for sharing. I will be testing the new firmware soon BUT before that I will be making my own PCB using the ESP32 devkit that’s available online.

I was looking at the new schematic you shared and it doesn’t have any jumpers. I also read through the ECU Rev11 guide but it isn’t clear if we have to put jumpers in the schematic?

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
The issue arises because to connect SD card you need to have pullups on GPIO2 and GPIO 12. GPIO2 needs to pulled down to enter programming mode so if you have a pullup resistor, you need to disconnect it and ground the GPIO2 pin to be able to program the devkit.
GPIO12 is used by ESP32 to determine internal flash voltage. On most devkit and ESP32 Wroom32 modules it needs to be pulled low at startup. If it is pulled to VCC the ESP32 may not be able to read flash at all so it will keep resetting.
Since you are using devkit, you can solve both problems by programming the dev kit outside the main PCB. 
GPIO 2 grounding is already set in Dev kit so you can simply program it or in some cases you have to hold one of the buttons down to enter programming mode.
For GPIO12 you will have to find ESP_IDF folder and use the following command
components/esptool_py/esptool/espefuse.py set_flash_voltage 3.3V
This command will permanently set the flash voltage to 3.3V so GPIO12 state will have no effect on ESP32. This offcourse means use it only on modules which use 3.3v flash otherwise you may brick the module. This command cannot be changed once it is applied to the ESP32. All devkits that I have used are with 3.3v flash. Apparently Wrover module uses 1.8v flash and may be some other ESP32 modules.
You can also avoid this problem by ignoring sd circuit completely and not using resistors and sd module. The ECU works fine without sd module. It will check in the beginning and if it doesn’t find sd card it doesn’t try to record anything.

Regards



Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
Understood. So I will have to program the ESP32 outside the main circuit pcb every time. That means I cannot solder it to the pcb permanently. It will have to be on the header pins always. ic an do that.

Regarding the GPIO2 grounding I never had to hold any button during the programming so i think its already taken care of in the Dev Kit.

Is there a way to check if my Dev kit has 3.3v or 1.8v on the flash?

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
I am trying to integrate sd card module to my controller.
How and where do you run the command "components/esptool_py/esptool/espefuse.py set_flash_voltage 3.3V" to permanently set the GPIO12 pin to 3.3v.

Do i need to install some software to run this or this can be done through Arduino IDE?

**Comment by Jehanzeb1973** [comment]
You are right. I had a few DevKits which require me to hold the button down. But there were other versions which programmed without requiring to press any button.
I couldn’t find a way to figure out the flash voltage. One way I THINK is to just pull the GPIO12 high at reset time and read serial data. If it keeps giving you flash read error then it is a 3.3v flash. If it boots normally even while pulled high then it’s a 1.8v flash but then you don’t need to program using esptools anyways as it will work fine even while pulled high.
Regards

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
I think I am going to try the ECU without the SD card functionality for now since this will not have any effect on the functioning.
You said we can do it by just removing the card slot and all the resistors with it. So all the pins that SD Card slot was routed to will stay empty?

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
This is correct. ECU will work fine if SD card is not present or even if pins are not connected to anything. Also , it will work fine if you switch off the webserver from screen 1. You will not see any data if you switch off webserver but it does not affect ECU working.

Regards


Jehanzeb khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
Thanks for the info. I have removed the SD card module and all its connectors and 10k resistors from the schematic and has sent the PCB out for manufacturing.
Once I have it, i will be ready for testing it on my engine.

Are there any ECU settings that you recommend changing before I test it?

I advise we make a separate thread with all the ECU settings that need tweaking for someone who wants to test it on different engine.

Thanks

**Comment by Jehanzeb1973** [comment]
Great. I am looking forward to your feedback. 
One key thing to note is that the ECU does not make any assumptions about starter and fuel pump voltage for control. Since I tested with a 3S Lipo, all the default settings are close to what you would expect if you are using 3S battery and similar starter motor and fuel pump as I am. 
ECU logic will work fine for different voltages and motor/pump types but the startup may be fast/slow depending on motor and fuel pump.
If you are using a 2S battery I would expect fuel pump and starter to be slower so you need to reduce starter and acceleration delays to  make them respond faster.
On the other hand if you are using 4S battery then starter and fuel pump will accelerate faster than what I will see with a 3S battery. So with a 4S battery you will probably need to increase starter and acceleration delays so that they accelerate a bit slower. 
In my case these are settings
Batt Volt= 3S = 11.1 V
Starter motor = RS380 rated 12V RPM 18000
Fuel Pump = Make=Conrad- Rated 6V

I will start a separate thread on GitHub for ECU settings with different setups.
Regards
Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
My starter motor is a rather beefy one this time - 4076 2000kv BLDC motor (i am not taking any chances)
Fuel pump is what already came with the engine. (JetCentral rhino 170 propane start engine)

I have 2 options-

1. Running starter on 4s (14.8v) and running the Fuel pump on 2s battery

OR

2. Running both starter and Fuel pump on 2s battery (7.4v). (In this case I am little bit concerned about max starter rpm of just 2000x7.4 = 14800 rpm)

What do you suggest?

Thanks.

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng

Following options are possible;

1st Option - If you can, then use a 3S Lipo. For my testing with brushless motor, I used 2700 kv motor and fuel pump was a 6v fuel pump, both running off 3S LiPo with brushed ESC’s.
2nd Option- use two batteries. A 4S Lipo for starter and a 2S for fuel pump. I did that for many runs and it works fine since each battery is only powering its own component.
3rd Option- use 2S battery but you will know if starter can speed up the engine to self sustaining RPM so this may or may not work. To save motor keep the tests to short interval and feel the motor with hand to see if it is getting hot.
Regards

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Comment by Jehanzeb1973** [comment]
Hi @dstromengThis command is run before burning the ECU firmware otherwise esp32 will keep going into reset cycle. You need to use edp-idf command line to run this command Regards Jehanzeb Get Outlook for AndroidFrom: dstromeng ***@***.***>Sent: Saturday, June 15, 2024 6:58:32 PMTo: Jehanzeb1973/Ardu_ECU ***@***.***>Cc: Jehanzeb1973 ***@***.***>; Mention ***@***.***>Subject: Re: [Jehanzeb1973/Ardu_ECU] New RPM sensor using AMR magnetic sensor (Discussion #8)
Hi @Jehanzeb1973
I am trying to integrate sd card module to my controller.
How and where do you run the command "components/esptool_py/esptool/espefuse.py set_flash_voltage 3.3V" to permanently set the GPIO12 pin to 3.3v.
Do i need to install some software to run this or this can be done through Arduino IDE?

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
I was able to burn the efuse using ESP IDF.
However the sd card is not able to be initiated. Says SD card not available.

I have tried everything including checking wiring and everything.
